File: backend/ESys.Utilty/Defs/ErrorCode.cs
/*
 *        ┏┓   ┏┓+ +
 *       ┏┛┻━━━┛┻┓ + +
 *       ┃       ┃
 *       ┃   ━   ┃ ++ + + +
 *       ████━████ ┃+
 *       ┃       ┃ +
 *       ┃   ┻   ┃
 *       ┃       ┃ + +
 *       ┗━┓   ┏━┛
 *         ┃   ┃
 *         ┃   ┃ + + + +
 *         ┃   ┃    Code is far away from bug with the animal protecting
 *         ┃   ┃ +     神兽保佑,代码无bug
 *         ┃   ┃
 *         ┃   ┃  +
 *         ┃    ┗━━━┓ + +
 *         ┃        ┣┓
 *         ┃        ┏┛
 *         ┗┓┓┏━┳┓┏┛ + + + +
 *          ┃┫┫ ┃┫┫
 *          ┗┻┛ ┗┻┛+ + + +
 */

using System.ComponentModel;

namespace ESys.Utilty.Defs
{
#pragma warning disable 1591
    public static class ErrorCode
    {
        private const int UserBase = 1000;
        private const int ServiceBase = 2000;
        private const int ReportBase = 3000;
        private const int WorkflowBase = 4000;
        private const int ApprovalBase = 5000;
        private const int DeviceBase = 6000;
        private const int ImportBase = 7000;


        [Description("成功")]
        public const int NoError = 0;

        public static class User
        {
            [Description("密码错误")]
            public const int WrongPassword = UserBase + 1;
            [Description("用户不存在")]
            public const int NotExist = UserBase + 2;
            [Description("用户被禁用")]
            public const int Forbidden = UserBase + 3;
            [Description("令牌过期，请重新登录")]
            public const int TokenExpired = UserBase + 4;
            [Description("没有权限，请联系管理员")]
            public const int NotAuthorized = UserBase + 5;
            [Description("验证码无效")]
            public const int CodeError = UserBase + 6;
            [Description("密码不符合规则")]
            public const int NotValidPassword = UserBase + 7;
            [Description("用户已存在")]
            public const int AlreadyExist = UserBase + 8;
        }

        public static class Service
        {
            [Description("内部错误")]
            public const int InnerError = ServiceBase + 1;
            [Description("需要电子签名")]
            public const int NeedESign = ServiceBase + 2;
            [Description("数据过期")]
            public const int DataExpired = ServiceBase + 3;
            [Description("数据不存在")]
            public const int DataNotExist = ServiceBase + 4;
            [Description("无法禁用数据")]
            public const int DeactiveInvalid = ServiceBase + 5;
            [Description("数据无效")]
            public const int InvalidModel = ServiceBase + 6;
            [Description("重复电子签名")]
            public const int DuplicateESign = ServiceBase + 7;
        }

        public static class Report
        {
            [Description("报告类型错误")]
            public const int TypeError = ReportBase + 1;
            [Description("报告不存在")]
            public const int NotExist = ReportBase + 2;
        }

        public static class Workflow
        {
            [Description("版本错误")]
            public const int VersionError = WorkflowBase + 1;
            [Description("缺乏开始步骤")]
            public const int LackStartStep = WorkflowBase + 2;
            [Description("缺乏开始步骤端口")]
            public const int LackStartStepPort = WorkflowBase + 3;
            [Description("开始步骤重复")]
            public const int TooManyStartStep = WorkflowBase + 4;
            [Description("加载失败")]
            public const int LoadError = WorkflowBase + 5;
        }

        public static class Approval
        {
            [Description("拒绝")]
            public const int Denied = ApprovalBase + 1;
        }

        public static class Device
        {
            [Description("配置错误")]
            public const int ConfigError = DeviceBase + 1;
            [Description("连接错误")]
            public const int ConnectError = DeviceBase + 2;
            [Description("获取数据错误")]
            public const int GetDataError = DeviceBase + 3;
            [Description("序列号错误")]
            public const int SNError = DeviceBase + 4;
            [Description("数据错误")]
            public const int DataError = DeviceBase + 5;
            [Description("未选择文件")]
            public const int NotSelectFile = DeviceBase + 6;
        }
        public static class Import
        {
            [Description("类型错误")]
            public const int TypeError = ImportBase + 1;
            [Description("不存在")]
            public const int NotExist = ImportBase + 2;
            [Description("数据错误")]

            public const int DataError = ImportBase + 3;

        }
    }
#pragma warning restore 1591
}

File: frontend/src/defs/ErrorCode.ts
const ErrorCode = {
  NoError: 0,
  User: {
    WrongPassword: 1001,
    NotExist: 1002,
    Forbidden: 1003,
    TokenExpired: 1004,
    NotAuthorized: 1005,
    CodeError: 1006,
    NotValidPassword: 1007,
    AlreadyExist: 1008
  },
  Service: {
    InnerError: 2001,
    NeedESign: 2002,
    DataExpired: 2003,
    DataNotExist: 2004,
    DeactiveInvalid: 2005,
    InvalidModel: 2006,
    DuplicateESign: 2007
  },
  Report: {
    TypeError: 3001,
    NotExist: 3002
  },
  Workflow: {
    VersionError: 4001,
    LackStartStep: 4002,
    LackStartStepPort: 4003,
    TooManyStartStep: 4004,
    LoadError: 4005
  },
  Approval: {
    Denied: 5001
  },
  Device: {
    ConfigError: 6001,
    ConnectError: 6002,
    GetDataError: 6003,
    SNError: 6004,
    DataError: 6005,
    NotSelectFile: 6006
  },
  Import: {
    TypeError: 7001,
    NotExist: 7002,
    DataError: 7003
  }
};

export default ErrorCode;

export function GetPromptKeyFromCode(code: number): string {
  switch (code) {
    case 0:
	    return 'ErrorPrompt.NoError';
    case 1001:
	    return 'ErrorPrompt.User.WrongPassword';
    case 1002:
	    return 'ErrorPrompt.User.NotExist';
    case 1003:
	    return 'ErrorPrompt.User.Forbidden';
    case 1004:
	    return 'ErrorPrompt.User.TokenExpired';
    case 1005:
	    return 'ErrorPrompt.User.NotAuthorized';
    case 1006:
	    return 'ErrorPrompt.User.CodeError';
    case 1007:
	    return 'ErrorPrompt.User.NotValidPassword';
    case 1008:
	    return 'ErrorPrompt.User.AlreadyExist';
    case 2001:
	    return 'ErrorPrompt.Service.InnerError';
    case 2002:
	    return 'ErrorPrompt.Service.NeedESign';
    case 2003:
	    return 'ErrorPrompt.Service.DataExpired';
    case 2004:
	    return 'ErrorPrompt.Service.DataNotExist';
    case 2005:
	    return 'ErrorPrompt.Service.DeactiveInvalid';
    case 2006:
	    return 'ErrorPrompt.Service.InvalidModel';
    case 2007:
	    return 'ErrorPrompt.Service.DuplicateESign';
    case 3001:
	    return 'ErrorPrompt.Report.TypeError';
    case 3002:
	    return 'ErrorPrompt.Report.NotExist';
    case 4001:
	    return 'ErrorPrompt.Workflow.VersionError';
    case 4002:
	    return 'ErrorPrompt.Workflow.LackStartStep';
    case 4003:
	    return 'ErrorPrompt.Workflow.LackStartStepPort';
    case 4004:
	    return 'ErrorPrompt.Workflow.TooManyStartStep';
    case 4005:
	    return 'ErrorPrompt.Workflow.LoadError';
    case 5001:
	    return 'ErrorPrompt.Approval.Denied';
    case 6001:
	    return 'ErrorPrompt.Device.ConfigError';
    case 6002:
	    return 'ErrorPrompt.Device.ConnectError';
    case 6003:
	    return 'ErrorPrompt.Device.GetDataError';
    case 6004:
	    return 'ErrorPrompt.Device.SNError';
    case 6005:
	    return 'ErrorPrompt.Device.DataError';
    case 6006:
	    return 'ErrorPrompt.Device.NotSelectFile';
    case 7001:
	    return 'ErrorPrompt.Import.TypeError';
    case 7002:
	    return 'ErrorPrompt.Import.NotExist';
    case 7003:
	    return 'ErrorPrompt.Import.DataError';
    default:
      throw "not implement in GetPromptKeyFromCode";
  }
}
